From a6a124ad950206accd2e58ab31b9ab2c0bbd7d9c Mon Sep 17 00:00:00 2001
From: Jiafei Pan <Jiafei.Pan@nxp.com>
Date: Thu, 25 May 2023 17:41:07 +0800
Subject: [PATCH 51/64] configs: arm64: fix and unify sw mailbox irqs for
 imx8mp

SPI irqs currently used conflict with:

76 I2C5 - I2C-5 Interrupt
77 I2C6 - I2C-6 Interrupt

So let's use free SPI irq 150 and 151.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>
---
 configs/arm64/imx8m-harpoon-config-industrial.h | 7 ++++---
 configs/arm64/imx8mm-harpoon-rtos.h             | 2 ++
 configs/arm64/imx8mn-harpoon-rtos.h             | 2 ++
 configs/arm64/imx8mp-harpoon-rtos.h             | 8 +++++---
 4 files changed, 13 insertions(+), 6 deletions(-)

diff --git a/configs/arm64/imx8m-harpoon-config-industrial.h b/configs/arm64/imx8m-harpoon-config-industrial.h
index e5e9dfd1..91180b59 100644
--- a/configs/arm64/imx8m-harpoon-config-industrial.h
+++ b/configs/arm64/imx8m-harpoon-config-industrial.h
@@ -21,8 +21,8 @@
 /* GIC */
 #define CONFIG_INMATE_IRQCHIPS_ADDR                   \
 	CONFIG_RTOS_IRQCHIPS_ADDR
-#define CONFIG_INMATE_IRQCHIPS 	                      \
-	IRQCHIP0(CONFIG_INMATE_IRQCHIPS_ADDR,                     \
+#define CONFIG_INMATE_IRQCHIPS 	                                         \
+	IRQCHIP0(CONFIG_INMATE_IRQCHIPS_ADDR,                            \
 		/* interrupts 32..63 */                                  \
 		RTOS_COMMON_IRQCHIP0_BITMAP1 |                           \
 		RTOS_INDUS_IRQCHIP0_BITMAP1,                             \
@@ -35,8 +35,9 @@
 		/* interrupts 128..159 */                                \
 		RTOS_COMMON_IRQCHIP0_BITMAP4 |                           \
 		RTOS_INDUS_IRQCHIP0_BITMAP4),                            \
-	IRQCHIP1(CONFIG_INMATE_IRQCHIPS_ADDR,                     \
+	IRQCHIP1(CONFIG_INMATE_IRQCHIPS_ADDR,                            \
 		/* interrupts 160..191 */                                \
+		RTOS_COMMON_IRQCHIP1_BITMAP1 |                           \
 		RTOS_INDUS_IRQCHIP1_BITMAP1,                             \
 		/* interrupts 192..223 */                                \
 		0,                                                       \
diff --git a/configs/arm64/imx8mm-harpoon-rtos.h b/configs/arm64/imx8mm-harpoon-rtos.h
index b00f2254..486f048d 100644
--- a/configs/arm64/imx8mm-harpoon-rtos.h
+++ b/configs/arm64/imx8mm-harpoon-rtos.h
@@ -126,6 +126,8 @@ typedef enum IRQn {
 		IRQnMAP(ENET_1588_IRQn)
 
 /* interrupts 160..191 */
+#define RTOS_COMMON_IRQCHIP1_BITMAP1 \
+	0
 #define RTOS_AUDIO_IRQCHIP1_BITMAP1 \
 	0
 #define RTOS_AVB_IRQCHIP1_BITMAP1 \
diff --git a/configs/arm64/imx8mn-harpoon-rtos.h b/configs/arm64/imx8mn-harpoon-rtos.h
index e048cbe1..990cb7e3 100644
--- a/configs/arm64/imx8mn-harpoon-rtos.h
+++ b/configs/arm64/imx8mn-harpoon-rtos.h
@@ -126,6 +126,8 @@ typedef enum IRQn {
 		IRQnMAP(ENET_1588_IRQn)
 
 /* interrupts 160..191 */
+#define RTOS_COMMON_IRQCHIP1_BITMAP1 \
+	0
 #define RTOS_AUDIO_IRQCHIP1_BITMAP1 \
 	0
 #define RTOS_AVB_IRQCHIP1_BITMAP1 \
diff --git a/configs/arm64/imx8mp-harpoon-rtos.h b/configs/arm64/imx8mp-harpoon-rtos.h
index 2fdade78..e373cdf5 100644
--- a/configs/arm64/imx8mp-harpoon-rtos.h
+++ b/configs/arm64/imx8mp-harpoon-rtos.h
@@ -21,8 +21,6 @@ typedef enum IRQn {
   I2S3_IRQn                    = 82,               /**< SAI3 Receive / Transmit Interrupt */
   GPT2_IRQn                    = 86,               /**< OR of GPT Rollover interrupt line, Input Capture 1 and 2 lines, Output Compare 1, 2, and 3 Interrupt lines */
   GPT1_IRQn                    = 87,               /**< OR of GPT Rollover interrupt line, Input Capture 1 and 2 lines, Output Compare 1, 2, and 3 Interrupt lines */
-  SW_MB_L_IRQn                 = 108,              /**< Unused SPI 76, to be used by generic software mailbox in Linux */
-  SW_MB_R_IRQn                 = 109,              /**< Unused SPI 77, to be used by generic software mailbox in RTOS */
   I2S56_IRQn                   = 122,              /**< SAI5/6 Receive / Transmit Interrupt */
   I2S1_IRQn                    = 127,              /**< SAI1 Receive / Transmit Interrupt */
   I2S2_IRQn                    = 128,              /**< SAI2 Receive / Transmit Interrupt */
@@ -31,6 +29,8 @@ typedef enum IRQn {
   ENET_QOS_IRQn                = 167,              /**< ENET QOS TSN LPI RX exit/Host System/RX/TX Channels[4:0] Interrupt */
   CAN_FD1_IRQn                 = 174,              /**< CAN-FD1 Interrupt from bus off/line error/RX warning/TX warning/wakeup/match in PN/timeout in PN/busoff done/FD error */
   CAN_FD1_ERROR_IRQn           = 175,              /**< CAN-FD1 Interrupt from correctable error/non correctable error int host/ non correctable error int internal */
+  SW_MB_L_IRQn                 = 182,              /**< Unused SPI 150, to be used by generic software mailbox in Linux */
+  SW_MB_R_IRQn                 = 183,              /**< Unused SPI 151, to be used by generic software mailbox in RTOS */
 } IRQn_Type;
 
 #define RTOS_COMMON_MEMORY_REGIONS \
@@ -114,7 +114,7 @@ typedef enum IRQn {
 
 /* interrupts 96..127 */
 #define RTOS_COMMON_IRQCHIP0_BITMAP3 \
-		IRQnMAP(SW_MB_R_IRQn)
+	0
 #define RTOS_AUDIO_IRQCHIP0_BITMAP3 \
 		IRQnMAP(I2S56_IRQn) | /* SAI5-6 */ \
 		IRQnMAP(I2S1_IRQn)    /* SAI1 */
@@ -135,6 +135,8 @@ typedef enum IRQn {
 	0
 
 /* interrupts 160..191 */
+#define RTOS_COMMON_IRQCHIP1_BITMAP1 \
+		IRQnMAP(SW_MB_R_IRQn)
 #define RTOS_AUDIO_IRQCHIP1_BITMAP1 \
 	0
 #define RTOS_AVB_IRQCHIP1_BITMAP1 \
-- 
2.34.1

